QENSmodels/delta_lorentz.py

```diff
@@ -74,16 +73,33 @@ def sqwDeltaLorentz(
 
     # Input validation
     q = np.asarray(q, dtype=np.float32)
+    A0 = np.asarray(A0)
+    hwhm = np.asarray(hwhm)
 
     # Create output array
     sqw = np.zeros((q.size, w.size))
 
     # Model
     if q.size > 1:
+        # if only a single float is given for A0, adapt to size of q
+        if A0.size == 1:
+            A0 = A0 * np.ones(q.size)
+        # else
+        # check that enough values of A0 are given to match the size of q
+        else:
+            assert A0.shape == q.shape, \
```

Comment:
will need to double-check that only object types with a `shape` attribute actually get to this part of the logic to be checked.